File: checks/check-partkey.pl
#!/usr/bin/perl
use Term::ANSIColor qw(:constants);
local $/;

$warningblocks = 5000;
$testwarning = 0;
$testerror = 0;

print "Checking participation key... ";

open (PART, "goal account partkeyinfo | grep 'Effective last round'|");
$part = <PART>;
$part =~ /Effective last round:\s+(\d+)/;
$part = $1;

open (STATUS, "goal node status | grep 'Last committed block'|");
$status = <STATUS>;
$status =~ /Last committed block:\s+(\d+)/;
$status = $1;

if ($testwarning)
{
  $part = $status + $warningblocks - 1;
}
if ($testerror)
{
  $part = $status - 10;
}

$diff = $part - $status;
#print "status = $status, part = $part, diff = $diff, warningblocks = $warningblocks\n";

if ($diff > $warningblocks)
{ 
  print GREEN "OK - $diff blocks until your key stops working\n" . RESET;
} 
elsif ($diff > 0)
{
  print YELLOW "WARNING - $diff blocks until your key stops working\n" . RESET;
}
else
{
  print RED "ERROR - Your key expired $diff blocks ago\n" . RESET;
}

File: checks/check-partkey.py
#!/usr/bin/python3
import re
from helper import *

warningblocks = 5000
testwarning = False
testerror = False
debug = False

printHEADER("Checking Participation Key")

# Goal participation key stats gathering 
mylastround = 0

goal = runCommand(['goal','node','lastround'])
thisround = int(goal)
if (thisround < 1):
  printERROR("Node not reporting last round")

goal = runCommand(['goal','account','partkeyinfo'])
goalcmd = re.search(r"Effective last round:\s+(\d+)",goal)
mylastround = 0
if goalcmd is not None:
   mylastround = int(goalcmd.group(1))
else:
   printERROR("Node not reporting Effective Last round")

diff = mylastround - thisround

if (testwarning):
  diff = warningblocks - 1
if (testerror):
  diff = -10

if (diff > warningblocks):
  printOK(str(diff) + " blocks until your key stops working")
elif (diff > 0):
  printWARNING(str(diff) + " blocks until your key stops working")
else:
  printERROR("Your key expired " + str(diff) + " blocks ago")
